app/test_admin.py

```diff
@@ -0,0 +1,208 @@
+
+        self.assertEqual(first=got_donor_id, second=self.donor.id)
+
+    def test_donor_contact_name(self) -> None:
+        got_donor_contact_name = self.donation_admin.donor_contact_name(
+            obj=self.donation)
+
+        self.assertEqual(first=got_donor_contact_name,
+                         second=self.donor.contact_name)
+
+    def test_donor_donor_name(self) -> None:
+        got_donor_donor_name = self.donation_admin.donor_donor_name(
+            obj=self.donation)
+
+        self.assertEqual(first=got_donor_donor_name,
+                         second=self.donor.donor_name)
+
+    def test_donor_email(self) -> None:
+        got_donor_email = self.donation_admin.donor_email(
+            obj=self.donation)
+
+        self.assertEqual(first=got_donor_email,
+                         second=self.donor.email)
+
+    def test_donor_mobile_number(self) -> None:
+        got_donor_mobile_number = self.donation_admin.donor_mobile_number(
+            obj=self.donation)
+
+        self.assertEqual(first=got_donor_mobile_number,
+                         second=self.donor.mobile_number)
+
+    def test_item_count(self) -> None:
+        want_count = len(self.donation.item_set.all())
+        got_count = self.donation_admin.item_count(obj=self.donation)
+
+        self.assertEqual(first=got_count, second=want_count)
+
+    def test_mark_items_verified(self) -> None:
+        request = self.request_factory.post(path="")
+        sessionMiddleware = SessionMiddleware()
+        sessionMiddleware.process_request(request=request)
+        messageMiddleware = MessageMiddleware()
+        messageMiddleware.process_request(request=request)
+        donations = Donation.objects.all()
+
+        self.donation_admin.mark_items_verified(req=request, qs=donations)
+
+        verified = self.donation.verified()
+
+        self.assertTrue(expr=verified)
+
+    def test_mark_items_unverified(self) -> None:
+        request = self.request_factory.post(path="")
+        sessionMiddleware = SessionMiddleware()
+        sessionMiddleware.process_request(request=request)
+        messageMiddleware = MessageMiddleware()
+        messageMiddleware.process_request(request=request)
+        donations = Donation.objects.all()
+
+        self.donation.item_set.update(verified=True)
+        self.donation_admin.mark_items_unverified(req=request, qs=donations)
+
+        verified = self.donation.verified()
+
+        self.assertFalse(expr=verified)
+
+    def test_generate_receipt(self) -> None:
+        self.donation.item_set.update(
+            status=ItemStatusEnum.RECEIVED, verified=True,
+            valuation_date="2023-09-12")
+        self.donation.valuation_date = "2023-09-12"
+        self.donation.save()
+
+        request = self.request_factory.post(path="")
+        request.user = self.user
+        donations = Donation.objects.all()
+
+        response = self.donation_admin.generate_receipt(
```

Comment:
Not here but I think the important thing to test here would be the views where we actually call `download_receipt` and `export_csv`. Please make sure to cover those in the future :) Otherwise LGTM

Reply:
Curious to see if it will be able to handle celery requirement as well

Reply:
<div><img src="https://media3.giphy.com/media/26H73Dl4HXXFe/giphy.gif?cid=5a38a5a2clxvx2nk5ckcnsht8ttemvguenv2hih6ddrak7m0&amp;ep=v1_gifs_search&amp;rid=giphy.gif&amp;ct=g" style="border:0;height:228px;width:300px"/><br/>via <a href="https://giphy.com/gifs/excited-excite-celery-26H73Dl4HXXFe">GIPHY</a></div>